core/src/main/scala/kafka/tools/StorageTool.scala

```diff
@@ -22,19 +22,37 @@ import java.nio.file.{Files, Paths}
 import kafka.server.{BrokerMetadataCheckpoint, KafkaConfig, MetaProperties, RawMetaProperties}
 import kafka.utils.{Exit, Logging}
 import net.sourceforge.argparse4j.ArgumentParsers
-import net.sourceforge.argparse4j.impl.Arguments.{store, storeTrue}
+import net.sourceforge.argparse4j.impl.Arguments.{store, storeTrue, append}
 import net.sourceforge.argparse4j.inf.Namespace
 import org.apache.kafka.common.Uuid
 import org.apache.kafka.common.utils.Utils
 import org.apache.kafka.metadata.bootstrap.{BootstrapDirectory, BootstrapMetadata}
-import org.apache.kafka.server.common.MetadataVersion
+import org.apache.kafka.server.common.{ApiMessageAndVersion, MetadataVersion}
+import org.apache.kafka.common.metadata.FeatureLevelRecord
+import org.apache.kafka.common.metadata.UserScramCredentialRecord
+import org.apache.kafka.common.security.scram.internals.ScramMechanism
+import org.apache.kafka.common.security.scram.internals.ScramFormatter
 
+
+import java.util
+import java.util.Base64
 import java.util.Optional
 import scala.collection.mutable
+import scala.jdk.CollectionConverters._
+import scala.collection.mutable.ArrayBuffer
 
 object StorageTool extends Logging {
   def main(args: Array[String]): Unit = {
     try {
+      main_internal(args)
```

Comment:
We don't use `snake_case`, we use `camelCase`.

Also, it seems messy to have calls to `Exit.exit` both inside the "internal" function and outside it. It makes you wonder why we're bothering with the internal function.

So what I'd sugest is:
1. rename `main_internal` to `run` or something like that
2. have `run` return the status code we should pass to `Exit.exit()`

Reply:
I want the String from the exception to validate that I go the correct error not just any. So instead I'll use the extended Exit.exit to capture the message. No more main_internal.